### PR TITLE
feat(STONEINTG-485): adding daily trigger and new workflow

### PR DESCRIPTION
**Pull request by jsztuka**
Trigger is scheduled the same as a current clair db build.
Workflow for building the db is being executed in RHTAP prod.

### CODE REVIEW

**Comment by jsztuka** [comment]
Neet do create clair-trigger branch

File: .github/workflows/clair-in-ci-db.yaml [hunk 1]
```diff
@@ -62,28 +46,24 @@ jobs:
...
 
           # check if required metadata and path to values are the same as expected
           jq -e '.data[].Features[0] | select(has("Name") and has("Vulnerabilities")) or error("Required keys do not exist")' results/clairdata.json
 
-      - name: Push-to-${{ env.REGISTRY }}
+      - name: Retag-and-push-to-${{ env.REGISTRY }}
         if: ${{ github.event_name != 'pull_request' }}  # don't push image from PR
         id: push-image
         run: |
+          docker tag ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.TO_TEST }} ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.LATEST_TAG }}
```
Comment (14rcole) [inline .github/workflows/clair-in-ci-db.yaml:68]: Should we also be retagging the previous image somehow in case we need to roll back?

Comment (jsztuka) [reply .github/workflows/clair-in-ci-db.yaml:68]: Something like this?
https://github.com/redhat-appstudio/hacbs-test/pull/115/files#diff-9b2343fac9b3b02fec14cb23c3fa1fce1410a3c1a40a1114ce0514e9f5c16419R44-R45

**Comment by jsztuka** [comment]
Branch created.

File: .tekton/build_and_trigger.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+              image: registry.access.redhat.com/ubi9/python-39
+              env:
+                - name: GITHUB_TOKEN
+                  valueFrom:
+                    secretKeyRef:
+                      name: integration-github-token
+                      key: integration-github-token
+              script: |
+                #!/usr/bin/env bash
+                curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GITHUB_TOKEN" --request POST --data '{"event_type":"trigger_workflow"}' https://api.github.com/repos/jsztuka/hacbs-test/dispatches
```
Comment (MartinBasti) [inline .tekton/build_and_trigger.yaml:71]: endpoint URL is wrong

**Comment by MartinBasti** [comment]
I'm not sure if I follow how it works:

Crontab:
trigger-clair-db-build.yaml --opens a PR ----> Opened PR ---triggers ---> build_and_trigger ---triggers-->  clair-in-ci-db.yaml  ---> tests and pushes image into latest

```
    pipelinesascode.tekton.dev/max-keep-runs: "3"
    pipelinesascode.tekton.dev/on-event: "[pull_request]"
    pipelinesascode.tekton.dev/on-target-branch: "[main]"
    pipelinesascode.tekton.dev/task: "[buildah]"
```
It has triggers for main and pull_request, how does it work, does it mean, that every PR and every update of PR will trigger rebuild of image from main branch?

PR:
How we are going to test changes in the PR? Build always uses main.
Why dockerlinter was removed without replacement?

Merge:
More or less works

File: .github/workflows/clair-in-ci-db.yaml [hunk 3]
```diff
@@ -1,44 +1,28 @@
...
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3
-
-      - name: Dockerfile linter
-        if: ${{ github.event_name == 'pull_request' }}  # don't break regular rebuilds if linter is updated
-        uses: hadolint/hadolint-action@v3.1.0
-        with:
-          dockerfile: clair-in-ci/Dockerfile
-          ignore: DL3041
```
Comment (MartinBasti) [inline .github/workflows/clair-in-ci-db.yaml:40]: Where are we linting Dockerfile now?

File: .github/workflows/clair-in-ci-db.yaml [hunk 4]
```diff
@@ -62,28 +46,24 @@ jobs:
...
-          docker run --rm -t ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.LATEST_TAG }} clair-action --version
+          docker pull ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.TO_TEST }}
+          docker run --rm -t ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.TO_TEST }} clair-action --version
 
       - name: Check Clair output format
         run: |
           # test real life usage of clair
           mkdir results
-          docker run --rm -v $(pwd)/results:/results:rw -t ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.LATEST_TAG }} bash -c 'clair-action report --image-ref=registry.access.redhat.com/ubi9-minimal --db-path=/tmp/matcher.db --format=quay > /results/clairdata.json'
+          docker run --rm -v $(pwd)/results:/results:rw -t ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.TO_TEST }} bash -c 'clair-action report --image-ref=registry.access.redhat.com/ubi9-minimal --db-path=/tmp/matcher.db --format=quay > /results/clairdata.json'
```
Comment (MartinBasti) [inline .github/workflows/clair-in-ci-db.yaml:58]: If we are using just floating tag TO_TEST, how can we be sure which image are we building, when there are multiple parallel builds? Should we use exact image digest instead and pass it as parameter to trigger?

File: .github/workflows/clair-in-ci-db.yaml (hunk 3 above)
Comment (jsztuka) [reply .github/workflows/clair-in-ci-db.yaml:40]: Right, this needs to stay.

**Comment by jsztuka** [comment]
My bad, I should have mentioned that in description.
Branch that is triggering PAC after the PR is created to it is currently **MAIN**. It was due to the fact to see if the PAC gets triggered correctly and all the included tasks are executing as they should.
You re correct with the fact that if we would merge this to the main, any newly created PR would trigger the PAC, which we don't want.
In order to prevent this behaviour, I will create dedicated branch - **PAC-trigger** which will be used for this purpose instead of main.

File: .github/workflows/clair-in-ci-db.yaml (hunk 4 above)
Comment (jsztuka) [reply .github/workflows/clair-in-ci-db.yaml:58]: Thats the thing, only one build and one build only must be triggered daily.

**Comment by MartinBasti** [comment]
So how changes in PRs will be tested?

**Comment by dirgim** [comment]
@jsztuka are we good to close this PR?

**Comment by jsztuka** [comment]
This has been resolved in new repo created specifically for clair-in-ci, to build daily db image.

**Pull request closed without merging**